pipes-schema-validation-confluent/README.md

```diff
@@ -0,0 +1,240 @@
+   kafka-confluent-validate.EnrichmentLambdafunctionname = kafka-confluent-validate-kafkaconfluentvaldiateA4B-ABC123
+   kafka-confluent-validate.EventBridgePipe = arn:aws:pipes:<AWS Region>:<AWS account ID>:pipe/pipe-123abc
+   kafka-confluent-validate.Targetservicebusname = kafkaconfluentvalidateEventBusABC123
+   ```
+
+## Configuring the EventBridge pipe
+
+EventBridge Pipes allows you to configure how message are consumed from your Kafka stream via several parameters. Adjust these parameters in the CDK stack by editing the [CfnPipe](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_pipes/CfnPipe.html) resource:
+
+- `maximum_batching_window_in_seconds`: the maximum length of a time to wait for events.
+- `batch_size`: the maximum number of records to include in each batch.
+- `starting_position`: the position in a stream from which to start reading.
+
+See [Pipes batching and concurrency](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-batching-concurrency.html) for details.
+
+## How it works
+
+![Pattern architecture](img/architecture.png "Pattern architecture")
+
+This pattern sets up an EventBridge pipe with the following key elements:
+
+1. The **source** of the pipe is a Kafka cluster deployed on Confluent Platform or Confluent Cloud. EventBridge Pipes reads messages from the Kafka stream in batches and provides these to the enrichment function as an event payload.
+
+1. The **enrichment** step of the pipe consists of a Lambda function that validates the incoming messages against Confluent Schema Registry and deserializes the message from Avro before returning it.
+
+1. The **target** of this pattern is an EventBridge custom event bus which is synchronously invoked by EventBridge Pipes with the events returned by the enrichment Lambda function. EventBridge Pipes supports a variety of other targets, including Lambda, AWS Step Functions, Amazon API Gateway, API destinations, and more.
+
+The Python enrichment function uses the [`confluent-kafka`](https://github.com/confluentinc/confluent-kafka-python) library for schema validation and Avro deserialization. It uses [Powertools for AWS Lambda (Python)](https://docs.powertools.aws.dev/lambda/python/latest/) to implement Serverless best practices such as logging, tracing, secrets handling, typing, and JSON schema validation for incoming requests.
+
+The function performs the following tasks:
+
+1. For each event in the batch passed to the function, it validates that all expected fields (such as `topic` , `key`, and `value`) are present using the [Powertools validation utility](https://docs.powertools.aws.dev/lambda/python/latest/utilities/validation/).
+
+1. In the messages polled by the EventBridge pipe from the Kafka stream, the `key` and `value` of the message are `base64` encoded. Therefore, for each message contained in the batch passed to the function, the function decodes both the `key` and the `value`.
```

Comment:
nit: I got a bit confused here because it seems that the keys are plain strings (`base64`) whereas value encoder is Avro. Correct? (I could only tell from reading the Lambda source code). If so, consider explaining the different schemas used in the intro and elaborate that the Pipe will always encode the `key` and `value` as `base64`, no matter which source schema is used.